DQM/HLTEvF/plugins/HLTObjectMonitorHI.cc

```diff
@@ -0,0 +1,916 @@
+//         Created:  Sun, 29 Oct 2018 17:00:00 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+#include <sys/time.h>
+#include <cstdlib>
+
+// user include files
+#include "DataFormats/Common/interface/TriggerResults.h"
+#include "DataFormats/HLTReco/interface/TriggerObject.h"
+#include "DataFormats/HLTReco/interface/TriggerEvent.h"
+#include "DataFormats/HLTReco/interface/TriggerTypeDefs.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include "HLTrigger/HLTcore/interface/HLTConfigProvider.h"
+
+#include "DQMServices/Core/interface/DQMStore.h"
+#include "DQMServices/Core/interface/MonitorElement.h"
+#include "DQMServices/Core/interface/DQMEDAnalyzer.h"
+
+#include <TMath.h>
+#include <TStyle.h>
+#include <TLorentzVector.h>
+
+#include <unordered_map>
+//
+// class declaration
+//
+
+//using namespace edm;
+using namespace trigger;
+using std::vector;
+using std::string;
+using std::unordered_map;
+
+class HLTObjectMonitorHI : public DQMEDAnalyzer {
+  struct hltPlot
+  {
+    
+    MonitorElement * ME;
+    string pathName;
+    string pathNameOR;
+    string moduleName;
+    string moduleNameOR;
+    int pathIndex = -99;
+    int pathIndexOR = -99;
+    string plotLabel;
+    string xAxisLabel;
+    int nBins;
+    double xMin;
+    double xMax;
+    bool displayInPrimary;
+    
+  };
+
+   public:
+      explicit HLTObjectMonitorHI(const edm::ParameterSet&);
+      ~HLTObjectMonitorHI() override;
+
+  //      static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+
+   private:
+      void analyze(const edm::Event&, const edm::EventSetup&) override;
+      void bookHistograms(DQMStore::IBooker &i, edm::Run const&, edm::EventSetup const&) override;
+      void dqmBeginRun(edm::Run const&, edm::EventSetup const&) override;
+      void endRun(edm::Run const&, edm::EventSetup const&) override;
+      vector<hltPlot*> plotList;
+      //virtual void beginLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+      //virtual void endLuminosityBlock(edm::LuminosityBlock const&, edm::EventSetup const&) override;
+  double get_wall_time(void);
+      // ----------member data ---------------------------
+
+  bool debugPrint;
+  HLTConfigProvider hltConfig_;
+  string topDirectoryName;
+  string mainShifterFolder;
+  string backupFolder;
+  unordered_map<string, bool> acceptMap;
+  unordered_map<hltPlot*, edm::ParameterSet*> plotMap;
+
+  //set Token(-s)
+  edm::EDGetTokenT<edm::TriggerResults>   triggerResultsToken_;
+  edm::EDGetTokenT<trigger::TriggerEvent> aodTriggerToken_;
+
+  //declare params
+  //2018 PbPb run
+  edm::ParameterSet HIPUCaloJet40Pt_pset;
```

Comment:
Instead of having one member variable for each PSet, you could just store them as values in the `plotMap`.
That is, change the type of `plotMap` from
```
unordered_map<hltPlot*, edm::ParameterSet*>
```
to
```
unordered_map<hltPlot*, edm::ParameterSet>
```
and move in there the PSets that you gfet from the configuration.